src/drivers/intel/cavs/hda-dma.c

```diff
@@ -403,8 +423,11 @@ static int hda_dma_link_copy_ch(struct dma *dma, struct hda_chan_data *chan,
 
 	/* make sure that previous transfer is complete (playback only) */
 	if (chan->direction == DMA_DIR_MEM_TO_DEV) {
-		while (hda_dma_get_free_size(dma, chan->index) < bytes)
-			idelay(PLATFORM_DEFAULT_DELAY);
+		int tx_err = hda_dma_tx_wait(dma, chan, bytes,
```

Comment:
minor, but variables should be declared on new lines at the start of the function (convention is to use err or ret). Fwiw, Safety critical has this requirement.